Reviewed 1 of 1 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

*[pkg/acceptance/terraform/nodectl, line 29 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/10628#-KWIvwNmaT-c5v41nMpE:-KWIvwNmaT-c5v41nMpF:b-mfu9vn) ([raw file](https://github.com/cockroachdb/cockroach/blob/4f1c27428d4171134fc82dd76fb2c2725adaee80/pkg/acceptance/terraform/nodectl#L29)):*
> ```
> gcs_url="$2"
> hostname="$(hostname)"
> if [ -z ${NODE_INDEX+x} ]; then
> ```

I believe this can be written `node_index=${NODE_INDEX-${hostname##*-}}`. Tested locally.

---

*[pkg/acceptance/terraform/nodectl, line 18 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/10628#-KWIxY8shC3w5iKUGT_-:-KWIxY8shC3w5iKUGT_0:brsly18) ([raw file](https://github.com/cockroachdb/cockroach/blob/a8f0b1becedbd336a9440e186f596e384afb3adf/pkg/acceptance/terraform/nodectl#L18)):*
> ```
> # Backup all regular files in the data directory.
> backup_data_dir() {
>     local backup_dir="${DATA_DIR}/backup-$(date +%Y%m%d-%H%M%S)"
> ```

this produces the following warning in shellcheck: `^-- SC2155: Declare and assign separately to avoid masking return values.`

might be worth running shellcheck over this script, or at least the parts you're adding.

---

*[pkg/acceptance/terraform/nodectl, line 21 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/10628#-KWIxfO0ZEN5WYq8e4qG:-KWIxfO0ZEN5WYq8e4qH:b-wvwr7o) ([raw file](https://github.com/cockroachdb/cockroach/blob/a8f0b1becedbd336a9440e186f596e384afb3adf/pkg/acceptance/terraform/nodectl#L21)):*
> ```
>     echo "Backing up files in ${DATA_DIR} to ${backup_dir}"
>     mkdir "${backup_dir}"
>     for f in "${DATA_DIR}"/*; do
> ```

perhaps `find "${DATA_DIR}" -type f | xargs -J % mv % "$backup_dir"`?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/10628)*
<!-- Sent from Reviewable.io -->

--- a/pkg/acceptance/terraform/nodectl
+++ b/pkg/acceptance/terraform/nodectl
@@ -13,6 +13,15 @@ usage() {
     exit 1
 }
 
+# Backup all regular files in the data directory.
+backup_data_dir() {
+    local backup_dir
+    backup_dir="${DATA_DIR}/backup-$(date +%Y%m%d-%H%M%S)"
+    echo "Backing up files in ${DATA_DIR} to ${backup_dir}"
+    mkdir "${backup_dir}"
+    find "${DATA_DIR}" -maxdepth 1 -type f -exec mv {} "${backup_dir}" \;
+}
+
 if [ $# -ne 2 ]; then
     echo "expected 2 parameters"
     echo ""
@@ -20,7 +29,7 @@ if [ $# -ne 2 ]; then
 fi
 
 # ~/.config/gcloud may be erroneously owned by root, so fix that.
-sudo chown -R ${USER} "${HOME}/.config/gcloud"
+sudo chown -R "${USER}" "${HOME}/.config/gcloud"
 
 # CockroachDB nodes are named blah-cockroach-[0-9]*, so extract the final
 # numeric part.
@@ -42,12 +51,11 @@ download)
         exit 1
     fi
     if [[ ! -z $(compgen -G "${DATA_DIR}/*.sst" 2>/dev/null) ]]; then
-        echo "sstables already exist in the destination"
-        exit 1
+        backup_data_dir
     fi
     echo "Downloading ${store_url} to store ${DATA_DIR}"
     cd "${DATA_DIR}"
-    gsutil cat ${store_url} | tar zxf -
+    gsutil cat "${store_url}" | tar zxf -
     ;;
 *)
     echo "Unknown action '${action}'"

--- a/pkg/acceptance/terraform/nodectl
+++ b/pkg/acceptance/terraform/nodectl
@@ -26,7 +26,7 @@ sudo chown -R ${USER} "${HOME}/.config/gcloud"
 # numeric part.
 gcs_url="$2"
 hostname="$(hostname)"
-node_index="${hostname##*-}" # remove everything through the final '-' in the hostname
+node_index=${NODE_INDEX-${hostname##*-}}
 store_url="${gcs_url}/store${node_index}.tgz"
 
 action=$1

